### PR TITLE
issue: 1480511 Fix TCP client hangs while connecting to local ip

### PR DESCRIPTION
**Pull request by liranoz12**
Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by liranoz12** [comment]
bot:retest

File: src/vma/sock/sockinfo_tcp.cpp [hunk 1]
```diff
@@ -3041,7 +3041,7 @@ int sockinfo_tcp::wait_for_conn_ready()
 		 * therefore in this case the m_conn_state will not be changed only
 		 * m_sock_state
 		 */
-		if (rx_wait(poll_count, m_b_blocking) < 0) {
+		if (rx_wait(poll_count, m_b_blocking) < 0 || unlikely(g_b_exit)) {
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: Please consider setting errno as done in other places
see:
https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_tcp.cpp#L626-L628
https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_tcp.cpp#L793-L797

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**